From 5c1ffac59c923cfd4aaa209440487615a655993d Mon Sep 17 00:00:00 2001
From: ouyanghui <ouyanghui@eswincomputing.com>
Date: Fri, 29 Aug 2025 11:16:06 +0800
Subject: [PATCH 517/540] fix:cpu frequency switching cause d2d protocol error

Changelogs:
1. add cpu aclk mux table, mux_u_cpu_aclk_2mux1_gfree.
2. add the switching process for u84_aclk and u84_busclk.
3. modify cpu-supported frequency
4. modify spll0_fout2 freq 1600MHz
5. split the opptable into eic7700-opptable.dtsi and eic7702-opptable.dtsi

Signed-off-by: ouyanghui <ouyanghui@eswincomputing.com>
---
 .../dts/eswin/eic7702-som560-interleave.dtsi  | 96 +++++--------------
 1 file changed, 25 insertions(+), 71 deletions(-)

diff --git a/arch/riscv/boot/dts/eswin/eic7702-som560-interleave.dtsi b/arch/riscv/boot/dts/eswin/eic7702-som560-interleave.dtsi
index 7b7b129bd0f5..dcec16c4d27b 100644
--- a/arch/riscv/boot/dts/eswin/eic7702-som560-interleave.dtsi
+++ b/arch/riscv/boot/dts/eswin/eic7702-som560-interleave.dtsi
@@ -507,7 +507,7 @@ &gpu0 {
 	status = "okay";
 };
 &d1_gpu {
-	status = "okay";
+	status = "disabled";
 };
 
 &gc820 {
@@ -719,59 +719,35 @@ &d1_graphcard2 {
 	status = "disabled";
 };
 
-&isp_0 {
-	status = "okay";
-};
-&d1_isp_0 {
+&vi_subsys {
 	status = "okay";
 };
 
-&isp_1 {
-	status = "okay";
-};
-&d1_isp_1 {
+&d1_vi_subsys {
 	status = "okay";
 };
 
-&dewarp {
-	status = "okay";
-};
-&d1_dewarp {
-	status = "okay";
+&isp_0 {
+	status = "disabled";
 };
 
-&mipi_dphy_rx {
-	status = "okay";
-};
-&d1_mipi_dphy_rx {
-	status = "okay";
+&d1_isp_0 {
+	status = "disabled";
 };
 
-&csi_dma0 {
-	status = "okay";
-};
-&d1_csi_dma0 {
-	status = "okay";
+&isp_1 {
+	status = "disabled";
 };
 
-&csi_dma1 {
-	status = "okay";
-};
-&d1_csi_dma1 {
-	status = "okay";
+&d1_isp_1 {
+	status = "disabled";
 };
 
-&csi2_0 {
-	status = "okay";
-};
-&d1_csi2_0 {
+&dewarp {
 	status = "okay";
 };
 
-&csi2_1 {
-	status = "okay";
-};
-&d1_csi2_1 {
+&d1_dewarp {
 	status = "okay";
 };
 
@@ -815,6 +791,7 @@ &d0_gmac0 {
 	rst-gpios = <&porta 29 GPIO_ACTIVE_LOW>;
 	eswin,rgmiisel = <&pinctrl 0x290 0x3>;
 	eswin,led-cfgs = <0x6100 0xa40 0x420>;
+	phy-mode = "rgmii-txid";
 	dly-param-1000m = <0x20202020 0x96206420 0x20202020>;
 	dly-param-100m = <0x20202020 0x96205A20 0x20202020>;
 	dly-param-10m = <0 0 0>;
@@ -827,6 +804,7 @@ &d1_gmac0 {
 	rst-gpios = <&d1_porta 29 GPIO_ACTIVE_LOW>;
 	eswin,rgmiisel = <&d1_pinctrl 0x290 0x3>;
 	eswin,led-cfgs = <0x6100 0xa40 0x420>;
+	phy-mode = "rgmii-txid";
 	dly-param-1000m = <0x20202020 0x8C205A20 0x20202020>;
 	dly-param-100m = <0x20202020 0x96205A20 0x20202020>;
 	dly-param-10m = <0 0 0>;
@@ -839,6 +817,7 @@ &d0_gmac1 {
 	rst-gpios = <&portb 2 GPIO_ACTIVE_LOW>;
 	eswin,rgmiisel = <&pinctrl 0x294 0x3>;
 	eswin,led-cfgs = <0x6100 0xa40 0x420>;
+	phy-mode = "rgmii-txid";
 	dly-param-1000m = <0x20202020 0x46205020 0x20202020>;
 	dly-param-100m = <0x20202020 0x50205020 0x20202020>;
 	dly-param-10m = <0 0 0>;
@@ -1181,11 +1160,11 @@ tps549d22@10 {
 		compatible = "ti,tps549d22";
 		reg = <0x10>;
 		eswin,regulator_default-microvolt=<1000000>;
-		eswin,regulator_label = "supply vdd1", "npu vdd1";
+		eswin,regulator_label = "Supply VDD", "D0 NPU Voltage";
 		label = "npu_vdd";
 		regulators{
 			d0_npu_vcc:npu_svcc{
-				regulator-name="NPU_SVCC";
+				regulator-name="D0_NPU_SVCC";
 				regulator-min-microvolt=<700000>;
 				regulator-max-microvolt=<1100000>;
 				regulator-min-microamp=<20000000>;
@@ -1203,11 +1182,11 @@ tps549d22@10 {
 		compatible = "ti,tps549d22";
 		reg = <0x10>;
 		eswin,regulator_default-microvolt=<1000000>;
-		eswin,regulator_label = "supply vdd1", "npu vdd1";
+		eswin,regulator_label = "Supply VDD1", "D1 NPU Voltage";
 		label = "d1_npu_vdd";
 		regulators{
 			d1_npu_vcc:npu_svcc{
-				regulator-name="NPU_SVCC2";
+				regulator-name="D1_NPU_SVCC";
 				regulator-min-microvolt=<700000>;
 				regulator-max-microvolt=<1100000>;
 				regulator-min-microamp=<20000000>;
@@ -1224,14 +1203,17 @@ &pwm0 {
 	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&pinctrl_die0_pwm0_default>;
 	pinctrl-1 = <&pinctrl_die0_gpio68_default>;
-	fan-gpios = <&portc 4 GPIO_ACTIVE_HIGH>;
+	fan-pwm-gpios = <&portc 4 GPIO_ACTIVE_LOW>;
+	fan-vcc-gpios = <&porta 18 GPIO_ACTIVE_LOW>;
 };
 &d1_pwm0 {
 	status = "okay";
 	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&pinctrl_die1_pwm0_default &pinctrl_die1_pwm1_default &pinctrl_die1_pwm2_default>;
 	pinctrl-1 = <&pinctrl_die1_gpio68_default &pinctrl_die1_gpio40_default &pinctrl_die1_gpio41_default>;
-	fan-gpios = <&d1_portc 4 GPIO_ACTIVE_HIGH>;
+	fan-pwm-gpios = <&d1_portc 4 GPIO_ACTIVE_LOW
+					&d1_portb 8 GPIO_ACTIVE_LOW
+					&d1_portb 9 GPIO_ACTIVE_LOW>;
 };
 
 &pvt0 {
@@ -1287,34 +1269,6 @@ &d1_rtc {
 	status = "okay";
 };
 
-&timer0 {
-	status = "okay";
-};
-&d1_timer0 {
-	status = "okay";
-};
-
-&timer1 {
-	status = "okay";
-};
-&d1_timer1 {
-	status = "okay";
-};
-
-&timer2 {
-	status = "okay";
-};
-&d1_timer2 {
-	status = "okay";
-};
-
-&timer3 {
-	status = "okay";
-};
-&d1_timer3 {
-	status = "okay";
-};
-
 &pinctrl {
 	status = "okay";
 };
-- 
2.54.0

